### ISSUE TITLE
Enforce client version through User-Agent header

### ISSUE THREAD

**Issue by sled** [outofcoffee/imposter#582]
I'm looking for a way to enforce compatibility between client versions and the mocked OpenAPI specification because developers tend to forget to update the mock server along with the client library in projects. We send the client library version in the User-Agent header and whenever the User-Agent header does not match a certain value I'd like to intercept the request and respond with a pre-defined status code.

I managed to do this with a security filter, however this limits the response code to 401.

```
security:
  conditions:
  - effect: Deny
    requestHeaders:
      user-agent:
        value: 'client/v2.0'
        operator: NotEqualTo
```

Another path I tried was to use a wildcard route, but this route does not take precedence over more specific routes.

```
  - path: /*
    requestHeaders:
      user-agent:
        value: 'client/v2.0'
        operator: NotEqualTo
    response:
      statusCode: 400
```

I could add a condition on the request header for every resource, but that's a bit repetitive and cumbersome. Is there an easy way to define some kind of middleware for the requests?

**Comment by outofcoffee** [comment]
#590 has been merged and released in [v3.43.0](https://github.com/outofcoffee/imposter/releases).

**Comment by sled** [comment]
awesome, thank you very much! this will be super useful.
